enhancements/api-review/add-ClusterImagePolicy-and-ImagePolicy-for-Signature-Verification.md

```diff
@@ -192,6 +196,46 @@ type FulcioCAWithRekor struct {
 	FulcioSubject PolicyFulcioSubject `json:"fulcioSubject,omitempty"`
 }
 
+// PKI defines the root of trust based on Root CA(s) and corresponding intermediate certificates.
+type PKI struct {
+	// CertificateAuthorityRootsData contains a certificate bundle PEM file containing one or more CA roots
+	// in the PEM format.
+	// caRootsData cannot be set if certificateChainData is set.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == '' || self.CertificateChainData == ''", message="CertificateAuthorityRoots cannot be set if CertificateChain is set"
+	CertificateAuthorityRootsData []byte `json:"caRootsData,omitempty"`
+
+	// CertificateAuthorityIntermediatesData contains a certificate bundle PEM file containing one or more intermediate certificates
+	// in the PEM format.
+	// caIntermediatesData cannot be set if certificateChainData is set.
+	// caIntermediatesData requires CertificateAuthorityRoots to be set.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self == '' || self.CertificateChainData == ''", message="caIntermediatesData cannot be set if certificateChainData is set"
+	// +kubebuilder:validation:XValidation:rule="self == '' || self.CertificateAuthorityRootsData != ''", message="caIntermediatesData requires caRootsData to be set"
+	CertificateAuthorityIntermediatesData []byte `json:"caIntermediatesData,omitempty"`
+
+	// CertificateIdentity must be set if any of the other fields are set.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="(self.CertificateChainData == '' && self.CertificateAuthorityRootsData == '' && self.caIntermediatesData == '') || self != ''", message="certificateIdentity must be set if any of certificateChainData, caRootsData, or caIntermediatesData are set"
+	// +kubebuilder:validation:XValidation:rule="self.Email != '' || self.Hostname != '' || self.IPAddress != ''", message="At least one of Email, Hostname, or IPAddress must be set in certificateIdentity"
+	CertificateIdentity *CertificateIdentity `json:"certificateIdentity,omitempty"`
+}
+
+// CertificateIdentity defines the identity type associated with the certificate identity
+type CertificateIdentity struct {
+	// Email address associated with the certificate identity.
+	// +optional
+	Email string `json:"email,omitempty"`
+
+	// Hostname associated with the certificate identity.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
+
+	// IP address associated with the certificate identity.
+	// +optional
+	IPAddress string `json:"ipaddress,omitempty"`
```

Comment:
I don’t know how IP address matching makes sense to administrators: “a certificate issued for 8.8.8.8 can sign our application”. It’s not _impossible_ but it seems all-around awkward to manage.

(… The Go certificate parser is already parsing IP addresses in SAN, so I guess adding this would not really introduce _risk_ and I wouldn’t block on this feature existing.)

Reply:
I agree, I will remove it.